cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +741,98 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath).replace("\r", ""))
```

Comment:
it seems wrong to do the 'replace' of \r here.  Either the content of that file was valid json/yaml or it wasnt.  A simple 'replace' is not going to know the difference between a \r in yaml and a \r as "windows end of line".

Reply:
fixed.